### PR TITLE
Added Images response in /reports and user roles in /project

### PR DESCRIPTION
**Pull request by EManahan**

### CODE REVIEW

File: billing-api/billing/__init__.py [hunk 1]
```diff
@@ -56,20 +57,25 @@ def login():
...
+                                  'roles': role_map[tenant.to_dict()['id']]},
+                  sessions.list_projects(client))
+
     return tenants
 
 
 @app.route('/reports', methods=['GET'])
 @authenticate
-def calculate_cost_by_user(client):
+def calculate_cost_by_user(client, user_id):
```
Comment (andricDu) [inline billing-api/billing/__init__.py]: Doesn't look like you ever use `user_id`

Comment (EManahan) [inline billing-api/billing/__init__.py]: Unsure of if there's a way to remove that parameter in the signature while still passing user_id to the function

Comment (EManahan) [inline billing-api/billing/__init__.py]: Planning on using user_id in /reports at a later date

File: billing-api/billing/auth/sessions.py [hunk 2]
```diff
@@ -2,6 +2,8 @@
 import json
 from keystoneclient.v2_0 import client
 from keystoneclient.exceptions import Unauthorized
+from keystoneauth1 import session
```
Comment (andricDu) [inline billing-api/billing/auth/sessions.py]: it doesn't look like you are using this import?

File: billing-api/billing/auth/sessions.py [hunk 3]
```diff
@@ -2,6 +2,8 @@
...
+from keystoneauth1.identity import V2Token
```
Comment (andricDu) [inline billing-api/billing/auth/sessions.py]: same goes for this import

**Comment by andricDu** [comment]
Actually, while we are still on this PR, I think it is time to factor out the defining of the bucket functions you have. 

Something like:
```python
def get_bucket_functions(bucket_size):
    if bucket_size == 'weekly':
    ...
  
    return same_bucket, next_bucket 
```

Then in your calculate cost function you can reduce the code to:
```python
same_bucket, next_bucket = get_bucket_functions(bucket_size)
```

**Comment by andricDu** [comment]
also I like that you explicitly show what the `else` case can be with 
```python
    bucket_size = 'daily'
```

but you might as well put that sucker as a comment
```python
    # bucket_size = 'daily'
```
or 
```python
    # Daily bucket size
```

**Comment by EManahan** [comment]
For the bucket_size = daily, I'm setting it to daily to cover the case where bucket_size is not set or an invalid value

**Comment by andricDu** [comment]
Makes sense, than maybe up before calling the function to get your bucketing functions you could do something like:

```python
bucket_size = 'daily' if bucket_size not in valid else bucket_size
```

Where valid can be defined as a constant like `['yearly', 'monthly', 'weekly', 'daily']`, maybe even in the configuration.

**Pull request merged**